OpenRA.Mods.Common/Traits/Attack/AttackGarrisoned.cs

```diff
@@ -182,8 +188,13 @@ public override void DoAttack(Actor self, Target target)
 					muzzleAnim.PlayThen(sequence, () => muzzles.Remove(muzzleFlash));
 				}
 
-				foreach (var npa in self.TraitsImplementing<INotifyAttack>())
-					npa.Attacking(self, target, a, barrel);
+				// Armament only notifies the Armament's owner, not the garrisoned actor, so we need to do this manually here
+				// to make traits like Cloak work on garrisoned actors.
+				// Thankfully, WithMuzzleOverlay is literally the only trait implementing INotifyAttack
+				// that actually uses the 'barrel' parameter, and AttackGarrisoned does its own muzzle stuff above,
+				// so we can safely pass null instead of doing more code gymnastics to get the correct barrel.
```

Comment:
My gut feeling is that we may regret this in the future. How about doing something like 744c6fdeb7ed5afa0ecc7bd03badb3c2792e7123 ?